homeassistant/components/system_health/__init__.py

```diff
@@ -101,6 +101,62 @@ async def get_integration_info(
     return result
 
 
+async def _registered_domain_data(
+    hass: HomeAssistant,
+) -> AsyncGenerator[tuple[str, dict[str, Any]]]:
+    registrations: dict[str, SystemHealthRegistration] = hass.data[DOMAIN]
+    for domain, domain_data in zip(
+        registrations,
+        await asyncio.gather(
+            *(
+                get_integration_info(hass, registration)
+                for registration in registrations.values()
+            )
+        ),
+        strict=False,
+    ):
+        yield domain, domain_data
+
+
+async def get_info(hass: HomeAssistant) -> dict[str, dict[str, str]]:
+    """Get the full set of system health information."""
+    domains: dict[str, dict[str, Any]] = {}
+
+    async def _get_info_value(value: Any) -> Any:
+        if not asyncio.iscoroutine(value):
+            return value
+        try:
+            return await value
+        except Exception as exception:  # noqa: BLE001
+            _LOGGER.error(
+                "Error fetching system info for %s - %s",
+                domain,
+                key,
+                exc_info=(type(exception), exception, exception.__traceback__),
+            )
+            return f"Failed: {exception.__class__.__name__}"
```

Comment:
I'd just include the string representation of the exception. It has a builtin magic `__str__` method.